fuel/transformers/__init__.py

```diff
@@ -2,18 +2,27 @@
 from multiprocessing import Process, Queue
 
 import numpy
-from picklable_itertools import chain, ifilter, izip, imap, repeat, starmap
-from picklable_itertools.extras import partition
+from picklable_itertools import chain, ifilter, izip
 from six import add_metaclass, iteritems
 
 from fuel import config
 from fuel.streams import AbstractDataStream
+from fuel.schemes import BatchSizeScheme
 
 
 @add_metaclass(ABCMeta)
 class Transformer(AbstractDataStream):
     """A data stream that wraps another data stream.
 
+    Subclasses must define a `transform_batch` method (to act on batches),
+    a `transform_example` method (to act on individual examples), or
+    both methods.
+
+    In all cases, the transformer is expected to have the same output
+    type (example or batch) as its input type.  If the transformer
+    subclass is going from batches to examples or vice versa, it
```

Comment:
I think I would also remove "In all cases," and say "Typically," but I feel this has lingered long enough. So I'm going to merge, and then send a separate pull request addressing your comment & mine.